src/commonTest/resources/dev/rhovas/interpreter/programs/rosettacode/Fibonacci.rho

```diff
@@ -0,0 +1,26 @@
+func pfib(n: Integer): Integer {
+    require n >= 0;
+
+    var lastFib = -1;
+    var fib = 1;
+    range(0, n, :incl).for {
```

Comment:
Apparently there's a typecheck issue with normal for loops and `range()`; I think my generics work regressed on this - will have to fix that.

Anyways, we do want to use a normal `for` loop here since the body is modifying local variables - it's better than using a lambda and it's likely semantic analysis checks in the future will affect this.

Reply:
So, semantically:

1. If I'm modifying variables outside of the for loop, it's better to do this:

`for (val i in range(1, n, :incl)) {...}`

2. If I'm NOT modifying variables outside of the for loop, it's better to do this:

`range(1, n, :incl).for {...}`

Am I understanding correctly? If so, what's the reasoning for this? Is it because lambdas are not as performant? I could form an argument that the lambda rendition is better for readability, especially since I'm not using i or val inside of the loop.

Also I am a Javascript goon so we can also note I have a bias for lambdas.

Reply:
Yep. It's mainly to avoid questions of unknown behavior down the road, though performance is relevant as well. Currently, since everything is interpreted it uses dynamic scope to Do The Right Thing. However, once this becomes compiled there are a lot of challenges related to closures and variable capturing (e.g. https://news.ycombinator.com/item?id=33160236). Those are hard problems to solve, so until I figure out what happens there it's better to avoid reliance on behavior there.

I have some ideas for annotations like @Immediate to indicate a lambda can only be run with the function it was passed to (and not stored somewhere to run later), but that comes way down the line.